app/ModelFunctions/PhotoFunctions.php

```diff
@@ -409,48 +356,86 @@ public function add(array $file, $albumID_in = 0)
 			Logs::notice(__METHOD__, __LINE__, $path_thumb);
 			$photo->thumbUrl = $path_thumb;
 
-			$resWidth = 0;
-			$resHeight = 0;
-
-			// Create Medium
-			if ($this->createMedium($photo, intval(Configs::get_value('medium_max_width')), intval(Configs::get_value('medium_max_height')), $resWidth, $resHeight)) {
-				$photo->medium = $resWidth . 'x' . $resHeight;
+			$this->createSmallerImages($photo);
+		}
 
-				if (Configs::get_value('medium_2x') === '1' &&
-				$this->createMedium($photo, intval(Configs::get_value('medium_max_width')), intval(Configs::get_value('medium_max_height')), $resWidth, $resHeight, true)) {
-					$photo->medium2x = $resWidth . 'x' . $resHeight;
-				}
-				else {
-					$photo->medium2x = '';
-				}
-			}
-			else {
-				$photo->medium = '';
-				$photo->medium2x = '';
-			}
+		return $this->save($photo, $albumID);
+	}
 
-			// Create Small
-			if ($this->createMedium($photo, intval(Configs::get_value('small_max_width')), intval(Configs::get_value('small_max_height')), $resWidth, $resHeight, false, 'SMALL')) {
-				$photo->small = $resWidth . 'x' . $resHeight;
+	/**
+	 * @param  Photo  $photo
+	 * @return void
+	 */
+	private function createSmallerImages(Photo $photo)
+	{
+		$mediumMaxWidth = intval(Configs::get_value('medium_max_width'));
+		$mediumMaxHeight = intval(Configs::get_value('medium_max_height'));
+		$this->resizePhoto($photo, 'medium', $mediumMaxWidth, $mediumMaxHeight);
 
-				if (Configs::get_value('small_2x') === '1' &&
-				$this->createMedium($photo, intval(Configs::get_value('small_max_width')), intval(Configs::get_value('small_max_height')), $resWidth, $resHeight, true, 'SMALL')) {
-					$photo->small2x = $resWidth . 'x' . $resHeight;
-				}
-				else {
-					$photo->small2x = '';
-				}
-			}
-			else {
-				$photo->small = '';
-				$photo->small2x = '';
-			}
+		if (Configs::get_value('medium_2x') === '1') {
+			$this->resizePhoto($photo, 'medium2x', $mediumMaxWidth * 2, $mediumMaxHeight * 2);
 		}
 
-		return $this->save($photo, $albumID);
+		$smallMaxWidth = intval(Configs::get_value('small_max_width'));
+		$smallMaxHeight = intval(Configs::get_value('small_max_height'));
+		$this->resizePhoto($photo, 'small', $smallMaxWidth, $smallMaxHeight);
+
+		if (Configs::get_value('small_2x') === '1') {
+			$this->resizePhoto($photo, 'small2x', $mediumMaxWidth * 2, $mediumMaxHeight * 2);
```

Comment:
I think these should be `$smallMaxWidth * 2` and `$smallMaxHeight *2`?

Reply:
Good catch !